src/sparse/impl/KokkosSparse_spiluk_numeric_impl.hpp

```diff
@@ -290,7 +290,10 @@ struct ILUKLvlSchedTP1NumericFunctor
 #else
       auto fact = L_values(k) * U_values(U_row_map(prev_row));
 #endif
-      L_values(k) = fact;
+      if ( my_team == 0 ) L_values(k) = fact; 
+
+      team.team_barrier();
+
       Kokkos::parallel_for( Kokkos::TeamThreadRange( team, U_row_map(prev_row)+1, U_row_map(prev_row+1) ), [&] ( const long kk ) {
```

Comment:
Could `U_row_map(prev_row)` be unsigned?  If so, then this might emit build warnings, since it would cast from unsigned to signed (`long`).  Also, `long` is almost never the right type for anything.  (It's not even 64 bits on some platforms.)

Reply:
Thanks @mhoemmen for pointing it out. I fixed it in the new commit.